generator/src/main/java/com/linkedin/pegasus/generator/TemplateSpecGenerator.java

```diff
@@ -155,6 +162,16 @@ public Collection<ClassTemplateSpec> getGeneratedSpecs()
     return _classTemplateSpecs;
   }
 
+  /**
+   * Set the regex name patterns for schemas which needs to skip deprecated fields when generating the specs
```

Comment:
Include a more functional explanation: "For any record schema with a fully-qualified name that matches any of the provided patterns, all its fields marked as deprecated will be skipped during spec generation."